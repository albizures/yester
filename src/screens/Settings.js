import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet, Linking, Platform } from 'react-native'
import Container from '../components/Container'
import TopBar from '../components/TopBar'
import { logOut, isAuthorized, subscriptionStatus } from '../utils/session'
import SettingsItem, { types } from '../components/SettingsItem'
import { translate } from '../components/Translate'
import { resetAnalytics, screen, track } from '../utils/analytics'
import withUser, { shapeContextUser } from '../components/withUser'

class Settings extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
    contextUser: PropTypes.shape(shapeContextUser).isRequired,
  }

  willFocusListener = null

  componentDidMount () {
    const { addListener } = this.props.navigation
    this.willFocusListener = addListener('willFocus', this.load)
  }

  componentWillUnmount () {
    this.willFocusListener.remove()
  }

  load = () => {
    screen('Settings', {})
  }

  onPressLanguage = () => {
    this.props.navigation.navigate('Language')
  }

  onPressNotifications = () => {
    this.props.navigation.navigate('Notifications')
  }

  onPressTerms = () => {
    track('Terms', {})
    Linking.openURL('https://www.yester.app/terms')
  }

  onPressAbout = () => {
    track('About', {})
    Linking.openURL('https://www.yester.app')
  }

  onPressManage = async () => {
    const { navigation } = this.props
    const currentStatus = await isAuthorized(this.props)
    track('Manage Subscription', { subscriptionStatus: currentStatus.tag })

    if (currentStatus !== subscriptionStatus.PRO) {
      return navigation.navigate('Subscription', { currentStatus })
    }
    Platform.OS === 'ios'
      ? Linking.openURL('https://apps.apple.com/account/subscriptions')
      : Linking.openURL('https://play.google.com/store/account/subscriptions')
  }

  onLogOut = async () => {
    const { navigation } = this.props
    await logOut()
    track('Log Out', {})
    resetAnalytics()
    navigation.navigate('Auth')
  }

  onPressFacebook = async () => {}

  render () {
    const topBar = <TopBar title='settings.title' />
    return (
      <Container topBar={topBar}>
        <View style={styles.container}>
          {/*
          <SettingsItem title={translate('settings.item.facebook')} type={types.TEXT}
            onPress={this.onPressFacebook} />
            */}
          <SettingsItem
            title={translate('settings.item.language')}
            type={types.CHEVRON}
            onPress={this.onPressLanguage}
          />
          {
            <SettingsItem
              title={translate('settings.item.notifications')}
              type={types.CHEVRON}
              onPress={this.onPressNotifications}
            />
          }
          <SettingsItem
            title={translate('settings.item.terms')}
            type={types.CHEVRON}
            onPress={this.onPressTerms}
          />
          <SettingsItem
            title={translate('settings.item.about')}
            type={types.CHEVRON}
            onPress={this.onPressAbout}
          />
          <SettingsItem
            title={translate('settings.item.subscription')}
            type={types.CHEVRON}
            onPress={this.onPressManage}
          />
          <SettingsItem title={translate('settings.item.logOut')} onPress={this.onLogOut} />
        </View>
      </Container>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 20,
  },
})

export default withUser(Settings)
