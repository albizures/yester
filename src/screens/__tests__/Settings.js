import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Button } from 'react-native'

import Settings from '../Settings'

const navigation = { navigate: jest.fn() }

describe('src/screens/ConfirmEmail.js', () => {
  it('should render', () => {
    const tree = renderer.create(<Settings navigation={navigation} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  describe('when button to home is pressed', () => {
    it('should navigate to Home screen', () => {
      const shallowSettings = shallow(<Settings navigation={navigation} />)
      const shallowTouchableHighlights = shallowSettings.find(Button)
      const shallowButtonToHome = shallowTouchableHighlights.at(0)

      shallowButtonToHome.prop('onPress')()

      expect(navigation.navigate).toHaveBeenCalledTimes(1)
      expect(navigation.navigate).toHaveBeenCalledWith('Home')
    })
  })
})
