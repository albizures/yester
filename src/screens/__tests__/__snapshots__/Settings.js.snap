// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/ConfirmEmail.js should render 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "marginTop": 40,
          "textAlign": "center",
        },
      ]
    }
  >
    SETTINGS
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        to Home
      </Text>
    </View>
  </View>
</View>
`;
